custom_components/smartthinq_sensors/wideq/device.py

```diff
@@ -1026,8 +1028,12 @@ def init_device_info(self):
 
             model_data = self._model_data
             if "Monitoring" in model_data and "Value" in model_data:
-                # this are old V1 model
-                self._model_info = ModelInfo(model_data)
+                if model_data["Info"]["productType"] == "AirPurifier":
+                    # this is v2 air purifier
+                    self._model_info = ModelInfoV2AC(model_data)
```

Comment:
Why this specific condition is required? Is not possible to determine model info in standard way? I would avoid if condition based on `productType`,

Reply:
My air purifier model data have "Monitoring" and "Value", so without this commit it will be recognized as v1. Then decode_snapshot() in ModelInfo is called but it's _data["Monitoring"]["type"] is "JSON", not equal to "THINQ2", so it will return a empty object.
Then I found if I make my air purifier as ModelInfoV2AC, it will work without any problem.

Reply:
I change a little bit the logic to be more generic.
Rebase your PR and verify that my last commit is ok for this condition.